mypy.ini

```diff
@@ -33,7 +36,23 @@ cache_dir=/dev/null
 
 [mypy-aiohttp.*]
 follow_imports=skip
+
 [mypy-black]
 # The following is because of `patch_click()`. Remove when
 # we drop Python 3.6 support.
 warn_unused_ignores=False
+
+[mypy-black.files]
+# Unfortunately tomli has deprecated strings and changed the API type
+# annotations to bytes all while strings still work. We still use strings
+# since it's unclear whether this will last (depends on how much the
+# TOML people like bare CR as a newline sequence I suppose). Anyway,
+# some versions of tomli still specify str as the type so mypy complains
+# that the type: ignore is unnecessary. Gosh I wish I wasn't telling
+# this story.
+#
+# See also: https://github.com/psf/black/pull/2408
+#           https://github.com/pypa/pip/pull/10238 (because black is
+#             following whatever they do)
+#           https://github.com/toml-lang/toml/issues/837
+warn_unused_ignores=False
```

Comment:
Now this is an exciting PR, good job!

I sense some frustration towards this particular ignore :smile: 
If you want you can remove all of this by going to `black.files.py` and replacing
```python
with open(path_config, encoding="utf8") as f:
    pyproject_toml = tomli.load(f)  # type: ignore  # due to deprecated API usage
```
with
```python
with open(path_config, encoding="utf8") as f:
    pyproject_toml = tomli.loads(f.read())
```

The behavior 100% the same, no need to worry about that.

Reply:
> The behavior will be 100% the same, no need to worry about that.

If this changes I'll be *very* annoyed, but I still trust you so alright.